### PR TITLE
sms integration with organisation-service

### PR DESCRIPTION
**Pull request by harshal-egov**

### CODE REVIEW

File: backend/organisation/src/main/java/org/egov/service/NotificationService.java [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        return localizedMessageMap.get(OrganisationConstant.ORGANISATION_NOTIFICATION_ENG_LOCALE_CODE + "|" + rootTenantId).get(msgCode);
+    }
+
+    /**
+     * Builds msg based on the format
+     *
+     * @param message
+     * @param userDetailsForSMS
+     * @return
+     */
```
Comment (vishalgupta-eGov) [inline backend/organisation/src/main/java/org/egov/service/NotificationService.java:201]: Declare constants for below type of Strings.
"{individualName}"

File: backend/organisation/src/main/java/org/egov/service/NotificationService.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    @Autowired
+    private HRMSUtils hrmsUtils;
+
+    /**
+     * Sends notification by putting the sms content onto the core-sms topic
+     *
+     * @param request
+     */
+    public void sendNotification(OrgRequest request, boolean isCreateOperation) {
+
```
[the remark this answers is not in the document]
Comment (harshal-egov) [reply backend/organisation/src/main/java/org/egov/service/NotificationService.java:55]: Keeping it as generic as possible to accomodate future changes as suggested by CK. If we use a single method we will have to change the strructure of the code again

File: backend/organisation/src/main/java/org/egov/service/NotificationService.java (hunk 1 above)
Comment (harshal-egov) [reply backend/organisation/src/main/java/org/egov/service/NotificationService.java:201]: Keeping it generic for now

File: backend/organisation/src/main/java/org/egov/service/NotificationService.java [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        for (Organisation organisation : organisations) {
+
+            //get orgName, ID, contactPerson, mobileNumber, cbo-url
+            log.info("get orgName, ID, contactPerson, mobileNumber, cbo-url");
+            Map<String, List<String>> orgDetails = getDetailsForSMS(organisation);
+
+            for (int i = 0; i < orgDetails.get("personNames").size(); i++) {
+
+                Map<String, String> smsDetails = new HashMap<>();
+
```
[the remark this answers is not in the document]
Comment (harshal-egov) [reply backend/organisation/src/main/java/org/egov/service/NotificationService.java:83]: Keeping it generic for now

**Review by vishalgupta-eGov — APPROVED** [review]

**Pull request merged**